app/integrations/google_workspace/google_drive.py

```diff
@@ -0,0 +1,331 @@
+        .execute()
+    )
+    # pylint: enable=no-member
+    return result["id"]
+
+
+@handle_google_api_errors
+def create_file(name, folder, file_type):
+    """Create a new file in Google Drive.
+        Options for 'file_type' are:
+
+            - "document": Google Docs
+            - "spreadsheet": Google Sheets
+            - "presentation": Google Slides
+            - "form": Google Forms
+            - "site": Google Sites
+
+    Args:
+        name (str): The name of the new file.
+        folder (str): The id of the folder to create the file in.
+        file_type (str): The type of the new file.
+
+    Returns:
+        str: The id of the new file.
+    """
+
+    mime_type = {
+        "document": "application/vnd.google-apps.document",
+        "spreadsheet": "application/vnd.google-apps.spreadsheet",
+        "presentation": "application/vnd.google-apps.presentation",
+        "form": "application/vnd.google-apps.form",
+        "site": "application/vnd.google-apps.site",
+    }
+
+    if file_type not in mime_type:
+        raise ValueError(f"Invalid file_type: {file_type}")
+
+    mime_type_value = mime_type[file_type]
+
+    service = get_google_service("drive", "v3")
+    result = (
+        service.files()
+        .create(
+            body={"name": name, "parents": [folder], "mimeType": mime_type_value},
+            supportsAllDrives=True,
+            fields="id",
+        )
+        .execute()
+    )
+    return result["id"]
+
+
+@handle_google_api_errors
+def get_file_by_name(name, folder):
+    """Get a file by name in Google Drive.
+
+    Args:
+        name (str): The name of the file to get.
+        folder (str): The id of the folder to search in.
+
+    Returns:
+        list: A list of files that match the name.
+    """
+    service = get_google_service("drive", "v3")
+    results = (
+        service.files()
+        .list(
+            pageSize=1,
+            supportsAllDrives=True,
+            includeItemsFromAllDrives=True,
+            corpora="drive",
+            q="trashed=false and name='{}'".format(name),
+            driveId=folder,
+            fields="files(appProperties, id, name)",
+        )
+        .execute()
+    )
+    return results.get("files", [])
+
+
+@handle_google_api_errors
+def list_folders_in_folder(folder):
+    """List all folders in a folder in Google Drive.
+
+    Args:
+        folder (str): The id of the folder to list.
+
+    Returns:
+        list: A list of folders in the folder.
+    """
+    service = get_google_service("drive", "v3")
+    results = (
+        service.files()
+        .list(
+            pageSize=25,
+            supportsAllDrives=True,
+            includeItemsFromAllDrives=True,
+            corpora="user",
+            q="parents in '{}' and mimeType = 'application/vnd.google-apps.folder' and trashed=false".format(
+                folder
+            ),
+            fields="nextPageToken, files(id, name)",
+        )
+        .execute()
+    )
```

Comment:
I think we'd need to do this in a loop and monitor for the `nextPageToken` to make sure we've got all requests.  Something like this (which might be totally invalid syntax based on writing it in a comment field 😅)
```suggestion
    page_token = None
    all_files = []
    while True:
        results = (
            service.files()
            .list(
                pageSize=25,
                supportsAllDrives=True,
                includeItemsFromAllDrives=True,
                corpora="user",
                q="parents in '{}' and mimeType = 'application/vnd.google-apps.folder' and trashed=false".format(
                    folder
                ),
                fields="nextPageToken, files(id, name)",
                pageToken=page_token,
            )
            .execute()
        )
        all_files.extend(results.get('files', []))
        page_token = results.get('nextPageToken')
        if not page_token:
            break
```